server/metrics.go

```diff
@@ -24,181 +22,168 @@ const (
+			Name: metricNotificationsTotalName,
+			Help: "Number of notifications sent"},
+			[]string{"platform", "type"}),
+		metricSuccess: prometheus.NewCounterVec(prometheus.CounterOpts{
+			Name: metricSuccessName,
+			Help: "Number of push success."},
+			[]string{"platform", "type"}),
+		metricSuccessWithAck: prometheus.NewCounterVec(prometheus.CounterOpts{
+			Name: metricSuccessWithAckName,
+			Help: "Number of push success that contains ackId."},
+			[]string{"platform", "type"},
+		),
+		metricDelivered: prometheus.NewCounterVec(prometheus.CounterOpts{
+			Name: metricDeliveredName,
+			Help: "Number of push delivered."},
+			[]string{"platform", "type"},
+		),
+		metricFailure: prometheus.NewCounterVec(prometheus.CounterOpts{
+			Name: metricFailureName,
+			Help: "Number of push errors."},
+			[]string{"platform", "type"}),
+		metricFailureWithReason: prometheus.NewCounterVec(prometheus.CounterOpts{
+			Name: metricFailureWithReasonName,
+			Help: "Number of push errors with reasons."},
+			[]string{"platform", "type", "reason"}),
+		metricRemoval: prometheus.NewCounterVec(prometheus.CounterOpts{
+			Name: metricRemovalName,
+			Help: "Number of device token errors."},
+			[]string{"platform", "reason"}),
+		metricBadRequest: prometheus.NewCounter(prometheus.CounterOpts{
+			Name: metricBadRequestName,
+			Help: "Request to push proxy was a bad request",
+		}),
+		metricAPNSResponse: prometheus.NewHistogram(prometheus.HistogramOpts{
+			Name: metricAPNSResponseName,
+			Help: "Request latency distribution",
+		}),
+		metricFCMResponse: prometheus.NewHistogram(prometheus.HistogramOpts{
+			Name: metricFCMResponseName,
+			Help: "Request latency distribution",
+		}),
+		metricNotificationResponse: prometheus.NewHistogramVec(prometheus.HistogramOpts{
+			Name: metricNotificationResponseName,
+			Help: "Notifiction request latency distribution"},
+			[]string{"platform"}),
+		metricServiceResponse: prometheus.NewHistogram(prometheus.HistogramOpts{
+			Name: metricServiceResponseName,
+			Help: "Request latency distribution",
+		}),
+	}
 
-func init() {
 	prometheus.MustRegister(
-		metricNotificationsTotal,
-		metricSuccess,
-		metricSuccessWithAck,
-		metricFailure,
-		metricFailureWithReason,
-		metricRemoval,
-		metricBadRequest,
-		metricAPNSResponse,
-		metricFCMResponse,
-		metricServiceResponse,
-		metricNotificationResponse,
+		m.metricNotificationsTotal,
+		m.metricSuccess,
+		m.metricSuccessWithAck,
+		m.metricFailure,
+		m.metricFailureWithReason,
+		m.metricRemoval,
+		m.metricBadRequest,
+		m.metricAPNSResponse,
+		m.metricFCMResponse,
+		m.metricServiceResponse,
+		m.metricNotificationResponse,
 	)
-}
 
-func NewPrometheusHandler() http.Handler {
-	return promhttp.Handler()
+	return m
 }
 
-func incrementNotificationTotal(platform, pushType string) {
-	if MetricsEnabled {
-		metricNotificationsTotal.WithLabelValues(platform, pushType).Inc()
-	}
+func (m *metrics) shutdown() {
```

Comment:
Is this only for the tests?

Reply:
Yes, but I can also include the `Server.Stop()`

Reply:
Ok, let's do that then.